### PR TITLE
add string starts_ and ends_with tests (and a test suite)

### PR DESCRIPTION
**Pull request by ArsenArsen**

### CODE REVIEW

File: tests/meson.build [hunk 1]
```diff
@@ -0,0 +1,12 @@
+gtest = dependency('gtest_main')
```
Comment (no92) [inline tests/meson.build:1]: While just `apt-get install`ing on CI is fine for this, I do wonder how dealing with the dep in a cbuildrt environment is supposed to happen.

Comment (ArsenArsen) [reply tests/meson.build:1]: it isn't, we don't have a test stage in xbstrap

when we do, as a host tool

File: meson.build [hunk 2]
```diff
@@ -47,9 +43,20 @@ endif
 
 frigg_dep = declare_dependency(include_directories: incl)
 
+if not get_option('build_tests').disabled()
```
Comment (64) [inline meson.build:46]: Isn't this outer if statement redundant?

Comment (ArsenArsen) [reply meson.build:46]: yes, making a mental note to re check when not half asleep

Comment (64) [reply meson.build:46]: LGTM otherwise

File: meson.build [hunk 3]
```diff
@@ -1,8 +1,4 @@
-# frigg is header-only, so we do not need a compiler.
-project('frigg')
-
-# We alreadys need C++ on the build system to build the analyzer.
-add_languages('cpp', native: true)
+project('frigg', 'cpp', default_options: ['cpp_std=c++20'])
```
Comment (avdgrinten) [inline meson.build:1]: I think the previous dance with `add_languages` was intentional: we need to be able to build frigg when no target system compiler is available.

**Review by avdgrinten — APPROVED** [review]
LGTM.

**Pull request merged**